File: GreenSignal.Mobile/app/src/main/java/com/example/greensignal/domain/model/response/Inspector.kt
package com.example.greensignal.domain.model.response

import android.telephony.PhoneNumberUtils
import com.example.greensignal.data.remote.dto.response.InspectorDto
import com.example.greensignal.data.remote.dto.response.InspectorStatus
import com.example.greensignal.data.remote.dto.response.SavedFileDto
import java.util.Date

data class Inspector(
    val id: String = "",
    val fio: String = "",
    val phone: String = "",
    val reviewStatus: InspectorStatus = InspectorStatus.Created,
    val createdAt: Date = Date(),
    val updatedAt: Date? = null,
    val number: Int = -1,
    val internalEmail: String = "",
    val lat: Double? = null,
    val lng: Double? = null,
    val lastLatLngAt: Date? = null,
    val certificateId: String = "",
    val schoolId: String = "",
    val certificateDate: Date? = null,
    val certificateFile: SavedFileDto? = null,
    val photoFile: SavedFileDto? = null,
    val signature: SavedFileDto? = null,
)

fun InspectorDto.toInspectorAccount(): Inspector {
    return Inspector(
        id = id,
        fio = fio,
        phone = formatPhoneNumber(phone),
        reviewStatus = reviewStatus,
        createdAt = createdAt,
        updatedAt = updatedAt,
        number = number,
        internalEmail = internalEmail,
        lat = lat,
        lng = lng,
        lastLatLngAt = lastLatLngAt,
        certificateId = certificateId,
        schoolId = schoolId,
        certificateDate = certificateDate,
        certificateFile = certificateFile,
        photoFile = photoFile,
        signature = signature
    )
}

private fun formatPhoneNumber(phoneNumber: String): String {
    val formattedPhoneNumber = PhoneNumberUtils.formatNumber(phoneNumber, "RU")
    return formattedPhoneNumber ?: phoneNumber
}

File: GreenSignal/Data/Models/Inspector.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Data.Models
{
    public class Inspector
    {
        [Key]
        [Display(Name = "Идентификатор")]
        public Guid Id { get; set; }

        #region User Info

        [MaxLength(255)]
        [Display(Name = "Ф.И.О.")]
        public string FIO { get; set; }

        [MaxLength(11)]
        [RegularExpression(@"^(7([0-6]|[8-9])[0-9]{9})$",
            ErrorMessage = "Неверный формат номера")]
        [Display(Name = "Номер телефона")]
        public string Phone { get; set; }

        [Display(Name = "Статус заявки")]
        public InspectorStatus ReviewStatus { get; set; }

        public string? TelegramUserId { get; set; }
        public string? TelegramChatId { get; set; }

        public Guid? PhotoFileId { get; set; }
        public SavedFile? PhotoFile { get; set; }

        [Display(Name = "Дата создания")]
        public DateTime CreatedAt { get; set; }

        [Display(Name = "Дата обновления")]
        public DateTime? UpdatedAt { get; set; }

        [Display(Name = "Номер")]
        public int Number { get; set; }

        #endregion

        #region Email

        [MaxLength(100)]
        [EmailAddress(ErrorMessage = "Введите корректный E-mail.")]
        [Display(Name = "E-mail")]
        public string InternalEmail { get; set; }

        [Display(Name = "Пароль от почты")]
        [MinLength(8)]
        public string Password { get; set; }

        #endregion

        #region Location

        [Display(Name = "Широта")]
        public double? Lat { get; set; }

        [Display(Name = "Долгота")]
        public double? Lng { get; set; }

        [Display(Name = "Последнее обновление местоположения")]
        public DateTime? LastLatLngAt { get; set; }

        #endregion

        #region Certificate

        [MaxLength(100)]
        [Display(Name = "ID сертификата")]
        public string CertificateId { get; set; }

        [DataType(DataType.Date)]
        [Display(Name = "Дата выдачи сертификата")]
        public DateTime? CertificateDate { get; set; }

        [Required]
        [Display(Name = "Сертификат")]
        public Guid CertificateFileId { get; set; }
        public SavedFile CertificateFile { get; set; }

        [Required]
        public string SchoolId { get; set; }

        public Guid? SignatureId { get; set; }
        public SavedFile? Signature { get; set; }

        #endregion

    }

    public enum InspectorStatus
    {
        [Display(Name = "Создан")]
        Created = 0,
        [Display(Name = "Подтверждение провалено")]
        VerificationFailed = 50,
        [Display(Name = "Активен")]
        Active = 100,
        [Display(Name = "Забанен")]
        Banned = 200
    }
}
